File: java/v2/hello/src/main/java/xyz/hoper/hello/Application.java
package xyz.hoper.hello;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.ConfigurableApplicationContext;
import xyz.hoper.hello.annotation.GrpcService;
import xyz.hoper.hello.service.GrpcLauncher;

import java.util.Map;

@SpringBootApplication
public class Application {
    public static void main(String[] args) {
        ConfigurableApplicationContext configurableApplicationContext = SpringApplication.run(Application.class, args);
        Map<String, Object> grpcServiceBeanMap =  configurableApplicationContext.getBeansWithAnnotation(GrpcService.class);
        GrpcLauncher grpcLauncher = configurableApplicationContext.getBean("grpcLauncher",GrpcLauncher.class);
        grpcLauncher.grpcStart(grpcServiceBeanMap);
    }
}

File: server/static/kotlin/v2/hello/src/main/kotlin/xyz/hoper/Application.kt
package xyz.hoper

import org.springframework.boot.SpringApplication
import org.springframework.boot.autoconfigure.SpringBootApplication
import xyz.hoper.annotation.GrpcService
import xyz.hoper.service.GrpcLauncher

@SpringBootApplication
class Application {
    fun main(args: Array<String>) {
        val configurableApplicationContext = SpringApplication.run(Application::class.java, *args)
        val grpcServiceBeanMap = configurableApplicationContext.getBeansWithAnnotation(GrpcService::class.java)
        val grpcLauncher: GrpcLauncher = configurableApplicationContext.getBean("grpcLauncher", GrpcLauncher::class.java)
        grpcLauncher.grpcStart(grpcServiceBeanMap)
    }
}